### PR TITLE
Support websocket server in docker

### LINKED ISSUE #1166: Create separate docker container for websocket server
Update the existing `Dockerfile`, `docket-compose.yml` and related helm chart configurations to support a separate docker container for the ws server.

The following ports should be exposed:
- 8546 - for the websockets
- 8547 - for prometheus and liveness related endpoints

### PR DESCRIPTION
**Pull request by isavov**
**Description**:

This PR modifies the Dockerfile and docker-compose definitions in order to support running a websocket instance of the `hedera-json-rpc-relay` as a separate container. 

* Modify dockerfile
* Update docker-compose.yaml

**Related issue(s)**:

Fixes #1166 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: helm-chart/ci/default-values.yaml [hunk 1]
```diff
@@ -1,9 +1,13 @@
 # Default values for json-rpc-relay in CI. Configured to testnet
 config:
-  CHAIN_ID: '0x128'
-  HEDERA_NETWORK: 'testnet'
-  MIRROR_NODE_URL: 'https://testnet.mirrornode.hedera.com'
+  CHAIN_ID: "0x128"
```
Comment (Nana-EC) [inline helm-chart/ci/default-values.yaml:3]: Q: why to double quotes?
Revert if it was accidental for consistency

File: helm-chart/ci/default-values.yaml [hunk 2]
```diff
@@ -1,9 +1,13 @@
...
+  HEDERA_NETWORK: "testnet"
+  MIRROR_NODE_URL: "https://testnet.mirrornode.hedera.com"
 
 image:
   registry: registry:5001
   tag: test
+
+resources:
+  requests:
+    cpu: 550m
```
Comment (Nana-EC) [inline helm-chart/ci/default-values.yaml:13]: Q" Is this still needed with the recent Helm chart update?

File: helm-chart/templates/configmap.yaml [hunk 3]
```diff
@@ -31,3 +31,9 @@ data:
   TIER_2_RATE_LIMIT: {{ .Values.config.TIER_2_RATE_LIMIT | quote }}
   TIER_3_RATE_LIMIT: {{ .Values.config.TIER_3_RATE_LIMIT | quote }}
   WEB_SOCKET_PORT: {{ .Values.config.WEB_SOCKET_PORT | quote }}
+  WEB_SOCKET_HTTP_PORT: {{ .Values.config.WEB_SOCKET_HTTP_PORT | quote }}
+  WS_CONNECTION_LIMIT: {{ .Values.config.WS_CONNECTION_LIMIT | quote }}
+  WS_CONNECTION_LIMIT_PER_IP: {{ .Values.config.WS_CONNECTION_LIMIT_PER_IP | quote }}
+  WS_SUBSCRIPTION_LIMIT: {{ .Values.config.WS_SUBSCRIPTION_LIMIT | quote }}
+  WS_MULTIPLE_ADDRESSES_ENABLED: {{ .Values.config.WS_MULTIPLE_ADDRESSES_ENABLED | quote }}
+  WS_MAX_CONNECTION_TTL: {{ .Values.config.WS_MAX_CONNECTION_TTL | quote }}
```
Comment (Nana-EC) [inline helm-chart/templates/configmap.yaml:39]: nit: preserve newline at end of file

File: .env.example [hunk 4]
```diff
@@ -34,12 +34,12 @@ WS_POLLING_INTERVAL=
...
 WS_CONNECTION_LIMIT=
 WS_MAX_CONNECTION_TTL=
 WS_CONNECTION_LIMIT_PER_IP=
 WS_SUBSCRIPTION_LIMIT=
 WS_MULTIPLE_ADDRESSES_ENABLED=
 ETH_BLOCK_NUMBER_CACHE_TTL_MS=
 ETH_GET_BALANCE_CACHE_TTL_MS=
-ETH_FEE_HISTORY_FIXED=
-WEB_SOCKET_METRICS_PORT=
+ETH_FEE_HISTORY_FIXED=
```
Comment (Nana-EC) [inline .env.example:45]: nit: preserve newline at end of file

File: Dockerfile [hunk 5]
```diff
@@ -31,4 +36,5 @@ RUN apt-get update && \
 USER node
 
 # Run
-ENTRYPOINT ["npm", "run", "start"]
+ENTRYPOINT ["npm", "run"]
+CMD ["start"]
```
Comment (Nana-EC) [inline Dockerfile:40]: nit: preserve newline at end of file

File: helm-chart/templates/deployment.yaml [hunk 6]
```diff
@@ -57,3 +57,65 @@ spec:
       tolerations:
         {{- toYaml . | nindent 8 }}
       {{- end }}
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app:  {{ template "json-rpc-relay.name" . }}-websocket
```
Comment (Nana-EC) [inline helm-chart/templates/deployment.yaml:65]: define a `json-rpc-relay.wsname` in the `_helpers.tpl` so you don't have to keep doing this.

File: helm-chart/templates/deployment.yaml [hunk 7]
```diff
@@ -57,3 +57,65 @@ spec:
...
+      - name: {{ .Chart.Name }}-ws
+        image: "{{ .Values.image.registry }}/{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        args: ["start:ws"]
+        env: 
+        - name: SUBSCRIPTIONS_ENABLED
+          value: "true" #SUBSCRIPTIONS_ENABLED set to "true" is a requirement for the websocket component
+        envFrom:
+        - configMapRef:
+            name: {{ include "json-rpc-relay.fullname" . }}
```
Comment (Nana-EC) [inline helm-chart/templates/deployment.yaml:103]: Does this mean the JSON RPC API and the websocket deployment share the same config? SHouldn't they be different?
I'm wondering if instead we should have a separate chart?

**Review by Nana-EC — CHANGES_REQUESTED** [review]
A few suggestions.
I think we should consider a separate chart for this to allow simpler maintenance (each chart only has what it needs) and easier understanding of how each scales together.

Another chart under a `/charts` dir similar to PR 1242 is what i'm thinking.

**Review by Nana-EC — CHANGES_REQUESTED** [review]
A few suggestions.
I think we should consider a separate chart for this to allow simpler maintenance (each chart only has what it needs) and easier understanding of how each scales together.

Another chart under a `/charts` dir similar to PR 1242 is what i'm thinking.

**Comment by isavov** [comment]
> A few suggestions. I think we should consider a separate chart for this to allow simpler maintenance (each chart only has what it needs) and easier understanding of how each scales together.
> 
> Another chart under a `/charts` dir similar to PR 1242 is what i'm thinking.

Thanks! I was thinking of something similar as mentioned in the Notes for reviewer section, and in light of #1242  and the changes it makes I'll rework this PR to address only the docker and docker-compose changes needed for websocket and create a separate chart in a new PR after #1242 is merged.

**Review by Nana-EC — APPROVED** [review]

**Review by AlfredoG87 — APPROVED** [review]
LGTM  👍

**Comment by Nana-EC** [comment]
@isavov #1253 is merged so you can add a new deployment resource for the `eth_subcribe` server

**Pull request merged**